### ISSUE TITLE
Improper list format

### ISSUE THREAD

**Issue by JL95741** [nautobot/pynautobot#65]
### Environment
* Python version: v3.7
* Nautobot version: v1.3.3

### Steps to Reproduce
data_list contains details on the device via webhook including hostname.

1. in Python:
nautobot = self.nautobot_obj
device_hostname = data_list.get('nethostname')
devices = nautobot.dcim.devices
device_obj = devices.filter(name=device_hostname)
interface_list = interfaces.filter(device=device_hostname)

2.run the following in python:
print(type(interface_list))
print(interface_list)

### Expected Behavior
<class 'list'>
['Ethernet1/1', 'Ethernet1/2']

### Observed Behavior
<class 'list'>
[Ethernet1/1, Ethernet1/2]

**Comment by jvanderaa** [comment]
@JL95741 does this make sense?

**Comment by lampwins** [comment]
@JL95741 this is normal, expected behavior. Your `interface_list` is a collection of `pynautobot.models.dcim.Interfaces` instances, not strings. I think the confusion, in this case, is that the `__repr__` of a `Record` in pynautobot is `str(self)` and `__str__` of `Interfaces` is the `display` attribute in the API, which is just the name of the interface.

So while it _looks_ like `print(interface_list)` is supposed to be printing a list of strings, it is, in fact printing a list of **object representations**.

If you wanted to get the list of names as strings, you could do something like this:
```
interface_names = [i.name for i in interfaces.filter(device=device_hostname)]
```

@nautobot/pynautobot, perhaps we can take this as feedback to change the `Record` `__repr__` to include the class name?

**Comment by jvanderaa** [comment]
Makes sense to make the change in fact. I'll add that to the next milestone. Would that be a breaking change? I don't think it would @lampwins

**Comment by lampwins** [comment]
I would not consider modification of a `__repr__` to be a breaking change.

**Comment by jvanderaa** [comment]
Closing for #67
